src/DataGrid.tsx

```diff
@@ -880,51 +871,61 @@ function DataGrid<R, SR, K extends Key>(
     return isDraggedOver ? selectedPosition.idx : undefined;
   }
 
-  function getSelectedCellProps(rowIdx: number): SelectedCellProps<R, SR> | undefined {
-    if (selectedPosition.rowIdx !== rowIdx) return;
+  function getGridTabIndex() {
+    const { idx, rowIdx } = selectedPosition;
+    return isCellWithinBounds(selectedPosition) &&
+      (idx <= lastFrozenColumnIndex || idx >= colOverscanStartIdx) &&
```

Comment:
Isn't `isCellWithinBounds` redundant here?
```suggestion
    return (idx <= lastFrozenColumnIndex || idx >= colOverscanStartIdx) &&
```